lib/matplotlib/backend_bases.py

```diff
@@ -3058,14 +3058,50 @@ def press_pan(self, event):
         if (event.button not in [MouseButton.LEFT, MouseButton.RIGHT]
                 or event.x is None or event.y is None):
             return
-        axes = [a for a in self.canvas.figure.get_axes()
-                if a.in_axes(event) and a.get_navigate() and a.can_pan()]
-        if not axes:
+
+        # get all relevant axes and group them by patch-visibility and zorder
+        all_axes = defaultdict(lambda: defaultdict(list))
+        for a in self.canvas.figure.get_axes():
+            if a.in_axes(event) and a.get_navigate() and a.can_pan():
+                all_axes[a.patch.get_visible()][a.zorder].append(a)
```

Comment:
So I'm wondering why we need to group the Axes? Can we not sort by reverse zorder, and pan all Axes down until the one that has a visible patch?

Reply:
yes, in the first place that would be the same... however in a later stage I need the groups for 2 things:
- to trigger all other transparent axes with higher or equal zorder
- to check if an axes is a twin of a visible-patch axes at the same zorder (it was the only way I could find to avoid triggering twin-axes that are at the same zorder but whose parent is effectively below the clicked axes)

... Also, I don't think that making groups here should be a performance issue